pyo3-ffi/src/cpython/lock.rs

```diff
@@ -8,6 +8,15 @@ pub struct PyMutex {
     pub(crate) _pin: PhantomPinned,
```

Comment:
This not part of the PR but since colesbury's clarification I believe this is unnecessary, so it can be removed.